README.md

```diff
@@ -34,95 +37,93 @@ See [update build process svg](build-process-mermaid.md)
+$ mkdir fcos
+$ cd fcos
 ```
 
-In the above commands we:
-
-- created the `./fcos` directory
-- set file ACLs so that the `builder` user (uid `1000` inside the container) can write files
-- set file ACLs so that new files get created with ACLs that allow the `builder` user
-- gave the directory an SELinux file context for sharing with containers
-- changed our working directory into `./fcos`
+### Define a bash alias to run cosa
 
 Now we'll define a bash function that we can use to call the assembler
-container:
+container.  There are a number of tweaks here on top of base `podman`.
+
+Note: *You should run this command as non-root*
+
+It's also fully supported to use `podman` as root, but some of the arguments
+here need to change for that.
 
 ```
-$ cosa() {
-    env | grep COREOS_ASSEMBLER
-    set -x # so we can see what command gets run
-    sudo podman run --rm -ti -v ${PWD}:/srv/ --userns=host --device /dev/kvm --name cosa \
-               ${COREOS_ASSEMBLER_PRIVILEGED:+--privileged}                                          \
-               ${COREOS_ASSEMBLER_CONFIG_GIT:+-v $COREOS_ASSEMBLER_CONFIG_GIT:/srv/src/config/:ro}   \
-               ${COREOS_ASSEMBLER_GIT:+-v $COREOS_ASSEMBLER_GIT/src/:/usr/lib/coreos-assembler/:ro}  \
-               ${COREOS_ASSEMBLER_CONTAINER_RUNTIME_ARGS}                                            \
-               ${COREOS_ASSEMBLER_CONTAINER:-quay.io/coreos-assembler/coreos-assembler:latest} $@
-    rc=$?; set +x; return $rc
+cosa() {
+   env | grep COREOS_ASSEMBLER
+   id=$(id -u)
+   set -x # so we can see what command gets run
+   podman run --rm -ti --security-opt label=disable --privileged \
+              --uidmap=${id}:0:1 --uidmap=0:1:${id} \
+              -v ${PWD}:/srv/ -v /dev/kvm:/dev/kvm -v /dev/fuse:/dev/fuse \
+              --tmpfs /tmp --tmpfs /var/tmp --name cosa \
+              ${COREOS_ASSEMBLER_CONFIG_GIT:+-v $COREOS_ASSEMBLER_CONFIG_GIT:/srv/src/config/:ro}   \
+              ${COREOS_ASSEMBLER_GIT:+-v $COREOS_ASSEMBLER_GIT/src/:/usr/lib/coreos-assembler/:ro}  \
+              ${COREOS_ASSEMBLER_CONTAINER_RUNTIME_ARGS}                                            \
+              ${COREOS_ASSEMBLER_CONTAINER:-quay.io/coreos-assembler/coreos-assembler:latest} $@
+   rc=$?; set +x; return $rc
 }
 ```
 
-**NOTE**: We're using `cosa` here as it is much easier to type than `coreos-assembler`.
-
 This is a bit more complicated than a simple alias, but it allows for
 hacking on the assembler or the configs and prints out the environment and
 the command that ultimately gets run. Let's step through each part:
 
-- `sudo podman run --rm -ti`: standard container invocation
+- `podman run --rm -ti`: standard container invocation
+- `--privileged`: Note we're running as non root, so this is still safe (from the host's perspective)
+- `--security-opt label:disable`: Similar to `--privileged`.
+- `--uidmap=${id}:0:1 --uidmap=0:1:${id}`: We need user namespaces configured for unprivileged mode
```

Comment:
- `--uidmap=container_uid:host_uid:amount` 

the first one maps our user from the host to root inside the container - makes sense.

the second one maps root in the container to user `1` on the host and it does this for `${id}` users? so if my user id is `1001` then `1002` in the container maps to `1001` on the host? I must be reading this wrong.

Reply:
I guess the real question is: what are you trying to accomplish with the 2nd `uidmap`